frontend/src/components/user/OpportunitySaveUserControl.tsx

```diff
@@ -93,6 +96,42 @@ export const OpportunitySaveUserControl = ({
 
   const { checkFeatureFlag } = useFeatureFlags();
   if (!checkFeatureFlag("savedOpportunitiesOn")) return null;
+
+  if (type === "icon") {
+    return (
+      <>
+        {user?.token ? (
+          <SaveIcon
```

Comment:
[nit] since these returned components are very similar, I wonder if we could move the differentiation into the <SaveButton> and avoid the repetition here. Not necessary at all, just a thought